src/services/WalletService/deterministic/useDeterministicWallet.ts

```diff
@@ -27,117 +40,152 @@ const useDeterministicWallet = (
     identity,
     'DeterministicWallet'
   );
-  const [shouldInit, setShouldInit] = useState(false);
-  const [service, setService] = useState<IDeterministicWalletService | undefined>(); // Keep a reference to the session in order to send
-  const [assetToQuery, setAssetToQuery] = useState(undefined as ExtendedAsset | undefined);
-  const [network, setNetwork] = useState(undefined as Network | undefined);
 
-  // Initialize DeterministicWallet and get the uri.
-  useEffect(() => {
-    if (!shouldInit || !assetToQuery || !network) return;
-    setShouldInit(false);
-    const dwService = DeterministicWalletService({
-      handleInitRequest: () =>
-        dispatch({
-          type: DWActionTypes.CONNECTION_REQUEST
-        }),
-      handleInit: (session: Wallet, asset: ExtendedAsset) =>
-        dispatch({
-          type: DWActionTypes.CONNECTION_SUCCESS,
-          payload: { session, asset }
-        }),
-      handleReject: (err: string) =>
-        dispatch({
-          type: DWActionTypes.CONNECTION_FAILURE,
-          error: {
-            code: DeterministicWalletReducer.errorCodes.SESSION_CONNECTION_FAILED,
-            message: err
-          }
-        }),
-      handleAccountsSuccess: () =>
-        dispatch({
-          type: DWActionTypes.GET_ADDRESSES_SUCCESS
-        }),
-      handleEnqueueAccounts: (accounts: DWAccountDisplay[]) => {
-        return dispatch({
-          type: DWActionTypes.ENQUEUE_ADDRESSES,
-          payload: { accounts }
+  const getAccounts = async (session: Wallet, dpaths: ExtendedDPath[]) => {
+    // Trezor wallet uses getMultipleAddresses for fetching multiple addresses at a time. Ledger doesn't have this functionality.
+    if (session.getMultipleAddresses) {
+      await session
+        .getMultipleAddresses(dpaths)
+        .then((accounts) => {
+          dispatch({
+            type: DWActionTypes.ENQUEUE_ADDRESSES,
+            payload: { accounts }
+          });
+        })
+        .catch((err) => {
+          dispatch({
+            type: DWActionTypes.GET_ADDRESSES_FAILURE,
+            error: { code: DeterministicWalletReducer.errorCodes.GET_ACCOUNTS_FAILED, message: err }
+          });
         });
-      },
-      handleAccountsUpdate: (accounts: DWAccountDisplay[], asset: ExtendedAsset) => {
-        return dispatch({
-          type: DWActionTypes.UPDATE_ACCOUNTS,
-          payload: { accounts, asset }
+    } else {
+      console.error(`[getAccounts]: Selected HD wallet type has no getMultipleAddresses method`);
+    }
+  };
+
+  const handleAccountsQueue = (
+    accounts: DWAccountDisplay[],
+    network: Network,
+    asset: ExtendedAsset
+  ) => {
+    const addresses = accounts.map(({ address }) => address);
+    const balanceLookup =
+      asset.type === 'base'
+        ? () => getBaseAssetBalancesForAddresses(addresses, network)
+        : () => getSingleTokenBalanceForAddresses(asset, network, addresses);
+
+    try {
```

Comment:
async patterns:
- when using `try/catch` prefer await.
- otherwise use `then/catch`.
Not both together.